src/api/routes/tx.ts

```diff
@@ -19,37 +19,42 @@ export function createTxRouter(db: DataStore): RouterWithAsync {
   const router = addAsync(express.Router());
 
   router.getAsync('/', async (req, res) => {
-    const limit = parseTxQueryLimit(req.query.limit ?? 96);
-    const offset = parsePagingQueryInput(req.query.offset ?? 0);
-
-    const typeQuery = req.query.type;
-    let txTypeFilter: TransactionType[];
-    if (Array.isArray(typeQuery)) {
-      txTypeFilter = parseTxTypeStrings(typeQuery as string[]);
-    } else if (typeof typeQuery === 'string') {
-      txTypeFilter = parseTxTypeStrings([typeQuery]);
-    } else if (typeQuery) {
-      throw new Error(`Unexpected tx type query value: ${JSON.stringify(typeQuery)}`);
-    } else {
-      txTypeFilter = [];
-    }
+    try {
+      const limit = parseTxQueryLimit(req.query.limit ?? 96);
+      const offset = parsePagingQueryInput(req.query.offset ?? 0);
+
+      const typeQuery = req.query.type;
+      let txTypeFilter: TransactionType[];
+      if (Array.isArray(typeQuery)) {
+        txTypeFilter = parseTxTypeStrings(typeQuery as string[]);
+      } else if (typeof typeQuery === 'string') {
+        txTypeFilter = parseTxTypeStrings([typeQuery]);
+      } else if (typeQuery) {
+        throw new Error(`Unexpected tx type query value: ${JSON.stringify(typeQuery)}`);
+      } else {
+        txTypeFilter = [];
+      }
 
-    const { results: txResults, total } = await db.getTxList({ offset, limit, txTypeFilter });
+      const { results: txResults, total } = await db.getTxList({ offset, limit, txTypeFilter });
 
-    // TODO: fix these duplicate db queries
-    const results = await Bluebird.mapSeries(txResults, async tx => {
-      const txQuery = await getTxFromDataStore(tx.tx_id, db);
-      if (!txQuery.found) {
-        throw new Error('unexpected tx not found -- fix tx enumeration query');
-      }
-      return txQuery.result;
-    });
-    const response: TransactionResults = { limit, offset, total, results };
-    const schemaPath = require.resolve(
-      '@blockstack/stacks-blockchain-sidecar-types/api/transaction/get-transactions.schema.json'
-    );
-    await validate(schemaPath, response);
-    res.json(response);
+      // TODO: fix these duplicate db queries
+      const results = await Bluebird.mapSeries(txResults, async tx => {
+        const txQuery = await getTxFromDataStore(tx.tx_id, db);
+        if (!txQuery.found) {
+          throw new Error('unexpected tx not found -- fix tx enumeration query');
+        }
+        return txQuery.result;
+      });
+      const response: TransactionResults = { limit, offset, total, results };
+      const schemaPath = require.resolve(
+        '@blockstack/stacks-blockchain-sidecar-types/api/transaction/get-transactions.schema.json'
+      );
+      await validate(schemaPath, response);
+      res.json(response);
+    } catch (error) {
+      logError('error getting tx', error);
+      res.status(500);
```

Comment:
What's the reason for the try/catch here? 500 errors should fall through to the default express error handler

Reply:
I added this when I was trying to understand what was happening with https://github.com/blockstack/stacks-blockchain/issues/1702, and figured it wouldn't hurt to leave it there.  I can remove if you think the default handling is enough

Reply:
Yes please remove them 👍